.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,69 @@
+name: CI tests
+
+on: pull_request
+
+env:
+  TYK_DB_REDISHOST: localhost
+  TYK_GW_STORAGE_HOST: localhost
+
+jobs:
+  test:
+    name: Go ${{ matrix.go }} Redis ${{ matrix.redis-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
```

Comment:
I think it makes sense to run this on a range of mongo, redis and go versions. You can pick the currently supported versions as the required check in the branch protection rules.

Reply:
For the first iteration, we can go with it I think. If we want in future, we can extend it.